pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -82,10 +82,14 @@ func ApplyLocalityLBSetting(
 		// Do not apply default failover when locality LB is disabled.
 	} else if enableFailover && (localityLB.Enabled == nil || localityLB.Enabled.Value) {
 		if len(localityLB.FailoverPriority) > 0 {
-			applyPriorityFailover(loadAssignment, wrappedLocalityLbEndpoints, proxyLabels, localityLB.FailoverPriority)
+			prioritiesAssigned := applyPriorityFailover(loadAssignment, wrappedLocalityLbEndpoints, proxyLabels, localityLB.FailoverPriority)
+			// if failover and failoverPriority both are present then apply failover on the lowest priority.
```

Comment:
not sure what does this mean

Reply:
Here the implementation is such that suppose there is both failoverPriority and failover applied, then failoverPriority is applied first. Suppose we have applied n priorities for 0 to n. Then the endpoints with Priority n is divided into n, n+1, n+2 and n+3 priority according to the regionalFailover. Do you have any other approach to solve this? @hzxuzhonghu ?

Reply:
Do we want regional failover to work for all priority levels individually instead according to you?